crates/locked-app/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+//! Spin internal application interfaces
+//!
+//! This crate contains interfaces to Spin application configuration to be used
+//! by crates that implement Spin execution environments: trigger executors and
+//! host components, in particular.
+
+#![deny(missing_docs)]
+
+pub mod locked;
+mod metadata;
+pub mod values;
+
+pub use async_trait::async_trait;
+pub use locked::Variable;
+pub use metadata::{MetadataExt, MetadataKey};
+
+/// MetadataKey for extracting the application name.
+pub const APP_NAME_KEY: MetadataKey = MetadataKey::new("name");
+/// MetadataKey for extracting the application version.
+pub const APP_VERSION_KEY: MetadataKey = MetadataKey::new("version");
+/// MetadataKey for extracting the application description.
+pub const APP_DESCRIPTION_KEY: MetadataKey = MetadataKey::new("description");
+/// MetadataKey for extracting the OCI image digest.
+pub const OCI_IMAGE_DIGEST_KEY: MetadataKey = MetadataKey::new("oci_image_digest");
+
+/// Type alias for a [`Result`]s with [`Error`].
+pub type Result<T> = std::result::Result<T, Error>;
+
+/// Errors returned by methods in this crate.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// An error propagated from the [`spin_core`] crate.
+    #[error("spin core error: {0:#}")]
```

Comment:
Maybe just `#[transparent]`?

Reply:
This was intended as no change to existing behaviour, but if @lann feels this would be a good change then I can bundle it in here for sure.

Reply:
Probably fine :shrug:

Reply:
Thanks, done